diff --git a/src/main/java/org/openjdk/jextract/clang/libclang/Index_h$shared.java b/src/main/java/org/openjdk/jextract/clang/libclang/Index_h$shared.java
--- a/src/main/java/org/openjdk/jextract/clang/libclang/Index_h$shared.java
+++ b/src/main/java/org/openjdk/jextract/clang/libclang/Index_h$shared.java
@@ -52,7 +52,7 @@
     public static final ValueLayout.OfDouble C_DOUBLE = (ValueLayout.OfDouble) Linker.nativeLinker().canonicalLayouts().get("double");
     public static final AddressLayout C_POINTER = ((AddressLayout) Linker.nativeLinker().canonicalLayouts().get("void*"))
             .withTargetLayout(MemoryLayout.sequenceLayout(java.lang.Long.MAX_VALUE, C_CHAR));
-    public static final ValueLayout.OfLong C_LONG = (ValueLayout.OfLong) Linker.nativeLinker().canonicalLayouts().get("long");
+    public static final ValueLayout C_LONG = (ValueLayout) Linker.nativeLinker().canonicalLayouts().get("long");
 
     static final boolean TRACE_DOWNCALLS = Boolean.getBoolean("jextract.trace.downcalls");

