diff --git a/src/main/java/org/openjdk/jextract/clang/libclang/Index_h.java b/src/main/java/org/openjdk/jextract/clang/libclang/Index_h.java
--- a/src/main/java/org/openjdk/jextract/clang/libclang/Index_h.java
+++ b/src/main/java/org/openjdk/jextract/clang/libclang/Index_h.java
@@ -33,9 +33,10 @@
 import java.util.*;
 import java.util.function.*;
 import java.util.stream.*;
-
 import static java.lang.foreign.ValueLayout.*;
 import static java.lang.foreign.MemoryLayout.PathElement.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
 
 public class Index_h extends Index_h$shared {
 
@@ -45,13 +46,27 @@
 
     static final Arena LIBRARY_ARENA = Arena.ofAuto();
 
+    static final SymbolLookup SYMBOL_LOOKUP;
 
     static {
-        System.loadLibrary("clang");
+        String osName = System.getProperty("os.name");
+        if (osName.startsWith("AIX")) {
+            String javaHome = System.getProperty("java.home");
+            String libName = javaHome + "/lib/libclang.a";
+            String clangVersion;
+            try {
+                clangVersion = Files.readString(Path.of(javaHome + "/conf/jextract/libclang.version"));
+            } catch (Exception e) {
+                throw new IllegalStateException("Failed to read libclang.version", e);
+            }
+            String sharedMember = "(libclang.so." + clangVersion.trim() + ")";
+            SYMBOL_LOOKUP = SymbolLookup.libraryLookup(libName + sharedMember, LIBRARY_ARENA);
+        } else {
+            String libName = osName.startsWith("Windows") ? "libclang" : "clang";
+            System.loadLibrary(libName);
+            SYMBOL_LOOKUP = SymbolLookup.loaderLookup().or(Linker.nativeLinker().defaultLookup());
+        }
     }
-
-    static final SymbolLookup SYMBOL_LOOKUP = SymbolLookup.loaderLookup()
-            .or(Linker.nativeLinker().defaultLookup());
 
     private static final int CXError_Success = (int)0L;
     /**
